tt_metal/common/logger.hpp

```diff
@@ -120,7 +120,11 @@ class Logger {
                 level_names[static_cast<std::underlying_type_t<Level>>(level)]);
             std::string type_str = fmt::format(fmt::fg(fmt::color::green), "{:>23}", type_names[type]);
             fmt::print(*fd, "{} | {} | ", type_str, level_str);
+#if FMT_VERSION < 100000
             fmt::print(*fd, fmt, std::forward<Args>(args)...);
+#else
+            fmt::print(*fd, fmt::runtime(fmt), std::forward<Args>(args)...);
```

Comment:
Can you clarify what changed between versions?

Reply:
https://fmt.dev/latest/api.html#compile-time-format-string-checks
In the latest fmt library (v10), compile-time format string checks are enabled by default on compilers that support C++20.
Since _fmt_ variable in the function is _char const*_ type, it cannot be checked in compile-time thus we should wrap it with _fmt::runtime_.
https://github.com/tenstorrent/tt-metal/blob/8289f9fe531cc4b3f6bcc17cbbc8f992b03f2a63/tt_metal/common/logger.hpp#L109

Reply:
Oh, good to know.. Thank you.